### PR TITLE
Start domains app

### PR DESCRIPTION
**Pull request by groovecoder**
May not merge yet, unless/until we have some UI to go along with this. But I'd like a review and sanity-check on what I have so far.
### Run tests
1. `python manage.py test`
### Spot-check
1. `python manage.py migrate` to add the `domains` models' tables to the db
2. `python manage.py runserver`
3. Add a [`DomainAuthorization` via the django admin site UI](http://127.0.0.1:8000/admin/domains/domainauthorization/): use a real domain that you really control
4. Add the `TXT` record to your domain's DNS ...
   - Hostname: the `DomainAuthorization.token` value
   - Record type: `TXT`
   - Value: the `DomainAuthorization.token` value
5. `python manage.py validate_domain_authorizations
6. Go back to the [`DomainAuthorization` via the django admin site UI](http://127.0.0.1:8000/admin/domains/domainauthorization/) and check that **Status** is now "valid" and **Validated** is now the appropriate DateTime value.

### CODE REVIEW

File: domains/apps.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from __future__ import unicode_literals
+
+from django.apps import AppConfig
+
+
+class DnsConfig(AppConfig):
```
Comment (groovecoder) [inline domains/apps.py]: Oops, need to rename this to `DomainsConfig` and `name = 'domains'`

File: domains/management/commands/validate_domain_authorizations.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from datetime import datetime
+
+from dnslib.dns import DNSRecord, DNSQuestion, QTYPE
+
+from django.core.management.base import BaseCommand
+
+from ...models import DomainAuthorization
+
+
+DNS_SERVER = '208.67.222.222'  # OpenDNS
```
Comment (groovecoder) [inline domains/management/commands/validate_domain_authorizations.py]: Should probably move this to an environment variable?

File: domains/migrations/0003_auto_20160203_1758.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# Generated by Django 1.9.2 on 2016-02-03 17:58
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('domains', '0002_auto_20160202_2029'),
```
Comment (groovecoder) [inline domains/migrations/0003_auto_20160203_1758.py:11]: Could squash all these into a single migration for the initial merge/commit of the app.

File: domains/models.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+from __future__ import unicode_literals
+import uuid
+
+from django.contrib.auth.models import User
+from django.db import models
+
+
+AUTHORIZATION_STATUS_CHOICES = (
```
Comment (groovecoder) [inline domains/models.py:8]: This list is taken from the [ACME protocol Authorization Object](https://letsencrypt.github.io/acme-spec/#rfc.section.5.3) status list.

File: domains/models.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    ('unknown', 'unknown'),
+    ('pending', 'pending'),
+    ('processing', 'processing'),
+    ('valid', 'valid'),
+    ('invalid', 'invalid'),
+    ('revoked', 'revoked')
+)
+
+AUTHORIZATION_TYPE_CHOICES = (
```
Comment (groovecoder) [inline domains/models.py:17]: Taken from the [ACME protocol Identifier Validation Challenges](https://letsencrypt.github.io/acme-spec/#identifier-validation-challenges) types.

**Comment by openjck** [comment]
r+

**Pull request merged**